src/discord_utils.py

```diff
@@ -31,6 +31,54 @@ async def try_set_pin(message: Message, pin_state: bool) -> None:
         print("Altering message pin state failed:", e)
 
 
+def build_filepath_for_attachment(message_id: int, attachment: Attachment) -> str:
+    """
+    Generate a unique filepath for a given attachment for the purposes of saving it to disk
+
+    Generates a filepath in the following format:
+    <attachment_directory>/<Discord message ID>.<attachment ID>.<file extension>
+
+    For example:
+    /home/user/busty/625891304148303894.625891304081063986.mp3
+
+    Filepaths for each song should be unique and the rest of the codebase assumes that they are.
+    A Discord message can contain multiple attachments, including the attachment ID as well.
+
+    Args:
+        message_id: The ID of the Discord message that included this attachment.
+        attachment: The attachment to build a filepath for.
+
+    Returns:
```

Comment:
maybe just personal preference but I feel like what it returns is implied by the description, docstrings are probably better short and sweet for smth like Busty